src/pages/Index.js

```diff
@@ -65,54 +120,164 @@ modalWithImage.setEventListeners();
+      // formValidators["edit-avatar-form"].disableButton();
+      // formValidators["edit-avatar-form"].resetForms();
+      editAvatarModal.close();
+    })
+    .catch((err) => {
+      console.error("Failed to update user avatar:", err);
+    })
+    .finally(() => {
+      editAvatarModal.renderLoading(false);
+    });
+}
 
-  ".cards__list"
+const editAvatarModal = new ModalWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
 );
+editAvatarModal.setEventListeners();
 
-cardSection.renderItems();
+profileAvatarContainer.addEventListener("click", () => {
+  editAvatarModal.open();
+});
 
 /*                                       */
-/*             Event Handlers            */
+/*          LIKE & DISLIKE               */
 /*                                       */
 
+function handleLikeClick(card) {
+  if (card.setIsLiked === true) {
+    api
+      .removeLike(card._id)
+      .then(() => {
+        card.setIsLiked = false;
+        // card.handleLikeIcon();
+      })
+      .catch(console.error);
+  } else {
+    api
+      .addLike(card._id)
+      .then(() => {
+        card.setIsLiked = true;
+        // card.handleLikeIcon();
+      })
+      .catch(console.error);
+  }
+}
+
 function handleImageClick(name, link) {
   modalWithImage.open(name, link);
 }
 
-const user = new UserInfo("#modal-title-input", "#modal-description-input");
+/*                                       */
+/*              ADD CARD                 */
+/*                                       */
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo({ name: data.title, description: data.description });
-  editModal.close();
+function handleAddCardSubmit(data) {
+  const name = data.title;
+  const link = data.link;
+  cardModal.renderLoading(true);
+  api
+    .createNewCard({ name, link })
+    .then((res) => {
+      const cardElement = createCard(res);
+      cardSection.addItem(cardElement);
+      cardModal.close();
+      addCardForm.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardModal.renderLoading(false);
+    });
 }
 
-function handleAddCardSubmit(userInfo) {
-  const name = userInfo.title;
-  const link = userInfo.link;
-  const newCard = renderCard({ name, link });
-  cardSection.addItem(newCard);
-  cardModal.close();
+/*                                       */
+/*            DELETE CARD                */
+/*                                       */
+
+const confirmDeleteModal = new ModalWithConfirm("#delete-modal");
+confirmDeleteModal.setEventListeners();
+
+function handleDeleteClick(card) {
+  confirmDeleteModal.open();
+  confirmDeleteModal.handleDelete(() => {
+    console.log(card);
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        console.log("Card deleted successfully");
+        confirmDeleteModal.close();
+        card.handleDeleteCard();
+      })
+      .catch((err) => {
+        console
+          .error("Error deleting card:", err)
```

Comment:
Needs correction: `console.error` doesn't return promise, so it can't have `finally` block. `finally` block should be after `catch` block in `deleteCard` promise chain